doc: add a Working with nRF52 user guide (#1968)
NordicPlayground/fw-nrfconnect-nrf: doc/nrf/ug_nrf52.rst

Introduction
************

The nRF52 Series of System-on-Chip (SoC) devices embed a powerful yet low-power Arm Cortex-M4 processor with our industry leading 2.4 GHz RF transceivers.
All of the nRF52 Series SoCs have support for Bluetooth 5 features, in addition to multi-protocol capabilities.

See `nRF52 Series`_ for the technical documentation on the nRF52 Series chips and associated kits.

Supported boards
================

All supported boards have a PCA number, which is used to uniquely identify each board.
The build system uses a combination of device part code and PCA number to identify a board name in the format ``nRF52xxx-PCAxxxxx``.
For example, for the :ref:`nRF52840 Development Kit (PCA10056) <zephyr:nrf52840_pca10056>`, the board name is ``nRF52840-PCA20056``, and you must use ``nrf52840_pca10056`` as target when building.

Devices in the nRF52 Series are supported by these boards in the Zephyr open source project and in |NCS|.
Refer to the PCA number for building projects for these boards and find a link to their descriptions.

.. list-table::
   :header-rows: 1

>    * - DK

Reviewer: Suggestions:
- A column 'Device' could make it clearer for new customers which DK is used for which SoC. Also, someone looking for nRF52832 won't find it in the table (only after following the link to Zephyr).
- A note below the table referring to using nRF52840 DK for nRF52811 and nRF52 DK for nRF52810, explaining that the 'bigger' chip is used to emulate the functionality of the 'smaller', similar to what we had in the old SDK: https://infocenter.nordicsemi.com/topic/sdk_nrf5_v16.0.0/nrf52811_user_guide.html

Author: Leave as is because the title is supported boards not supported chips.

In addition, you can run the :ref:`zephyr:bluetooth-samples` that are included from the Zephyr project.

For available libraries, see :ref:`lib_bluetooth_services` (|NCS|) and :ref:`zephyr:bluetooth_api` (Zephyr).

Enhanced ShockBurst
===================

.. include:: ug_esb.rst
   :start-after: esb_intro_start
   :end-before: esb_intro_end

See the :ref:`ug_esb` user guide for information about how to work with Enhanced ShockBurst.
To start developing, check out the :ref:`esb_prx_ptx` sample.

Near Field Communication
========================

Near Field Communication (NFC) is a technology for wireless transfer of small amounts of data between two devices that are in close proximity.
It makes transferring data fast and easy when devices are in close proximity.
The range of NFC is typically <10 cm.

|NCS| provides two protocol stacks for developing NFC applications: Type 2 Tag and Type 4 Tag.
These stacks are provided in binary format in the `nrfxlib`_ repository.
See :ref:`nrfxlib:nfc` for documentation about the NFC stacks.

The NFC stack requires the NFCT driver for nRF52 devices, which is available as part of `nrfx`_.

> The nrfx project is included in the |NCS| as a module of the Zephyr repository.

Author: ```suggestion
The nrfx repository is included in the |NCS| as a module of the Zephyr repository.
```

.. _ug_nrf52:

Working with nRF52
##################

> The |NCS| provides support for developing on a variety of nRF52 Series devices, using the respective development kits.

Reviewer 2: ```suggestion
The |NCS| provides support for developing on all nRF52 Series devices, and contains board definitions for all development kits and reference design hardware.
```

> The |NCS| provides support for developing on a variety of nRF52 Series devices, using the respective development kits.

Author: Sentence updated as suggested by Joel.

Introduction
************

The nRF52 Series is a family of multiprotocol System on Chip (SoC) devices that embed a powerful yet low-power Arm Cortex-M4 processor.

> All of the nRF52 Series SoCs have support for Bluetooth 5 features, in addition to multiprotocol capabilities (for example, Bluetooth Mesh and 2.4 GHz proprietary).

Reviewer 3: We should not start talking about 15.4 protocols at this point. That has to be added once we have proper support that in NCS

=================

See :ref:`zephyr:boards-arm` for a list of which boards are supported by Zephyr.
Check the ``nRF52*-PCA100*`` board descriptions for information about each board, including the board name to use when building for the respective board.

For example, for the :ref:`nRF52840 Development Kit (PCA10056) <zephyr:nrf52840_pca10056>`, use ``nrf52840_pca10056`` as board name when building.

Secure bootloader chain
=======================

nRF52 Series devices support a secure bootloader solution based on the chain of trust concept.

See :ref:`ug_bootloader` for more information and instructions on how to enable one or more bootloaders in your application.

Supported protocols
*******************

The main area of application for the nRF52 Series is Bluetooth Low Energy, but you can also use the proprietary ESB protocol, NFC, or USB.
The following sections give pointers on where to start when working with these protocols.

To test the general capabilities of the 2.4 GHz radio transceiver, use the :ref:`radio_test` sample.

Bluetooth Low Energy
====================

When you develop a Bluetooth Low Energy (LE) application, you must use the Bluetooth LE stack.

> This stack is split into two different layers: the Bluetooth LE Host and the Bluetooth LE Controller.

Reviewer 4: Correct. There is the Bluetooth Host and Bluetooth LE controller. Also the Host consists of several layers. So the term layer should be used correctly here.

"The Bluetooth software stack is split into two core components: the Bluetooth Host and the Bluetooth LE Controller."

The |NCS| uses Zephyr's implementation of the Bluetooth LE Host.
For the Bluetooth LE Controller, you can choose between the nRF Bluetooth LE Controller that is implemented by Nordic Semiconductor and Zephyr's open-source Bluetooth LE Controller.

> The :ref:`ug_ble_controller` user guide contains more information about the two available controllers and instructions for switching between them.

Reviewer 2: No need to repeat what is in the references.  Suggest replace whole para with:
The nRF Connect SDK Bluetooth stack is fully supported by Nordic for nRF52 series devices.

Reviewer 5: I think the brief explanation that we have in here is nice.  IMO, it's not repetitive, as you need to jump to a link to find more info.

> See :ref:`zephyr:bluetooth` for documentation on Zephyr's Bluetooth LE stack, including both host and controller.

Reviewer 2: ```suggestion
See this :ref:`zephyr:bluetooth` for documentation on the Bluetooth LE Host and open source LE Controller.
```

> See :ref:`zephyr:bluetooth` for documentation on Zephyr's Bluetooth LE stack, including both host and controller.

Reviewer 4: I'm not sure linking to the implementation of the controller is needed. We should only link to documentation describing features support and options.

@cvinayak  @carlescufi Do we have something like that for the zephyr controller?

> See :ref:`zephyr:bluetooth` for documentation on Zephyr's Bluetooth LE stack, including both host and controller.

Reviewer 4: @ru-fu Right. But we don't use the controller directly, all of the Bluetooth API is through the host.
The thing we need for the controller are how to configure features on/off, or enable/disable feature options.

> See :ref:`zephyr:bluetooth` for documentation on Zephyr's Bluetooth LE stack, including both host and controller.

Reviewer 6: This is just a todo. At some point we either include technical documentation on the controller's inner workings on that page, or just list its features and an overview of the functionality.

> For documentation about the nRF Bluetooth LE Controller and information on what variants of the controller support with chips, see :ref:`nrfxlib:ble_controller`.

Reviewer 7: ```suggestion
For documentation about the nRF Bluetooth LE Controller and information on what variants of the controller support which chips, see :ref:`nrfxlib:ble_controller`.
```

The |NCS| contains a variety of :ref:`ble_samples` that target nRF52 Series devices.

> In addition, you can run any of Zephyr's :ref:`zephyr:bluetooth-samples`.

Reviewer 2: ```suggestion
In addition, you can run the ref:`zephyr:bluetooth-samples` included from the Zephyr project.
```

For available libraries, see :ref:`lib_bluetooth_services` (|NCS|) and :ref:`zephyr:bluetooth_api` (Zephyr).

Enhanced ShockBurst
===================

.. include:: ug_esb.rst
   :start-after: esb_intro_start
   :end-before: esb_intro_end

See the :ref:`ug_esb` user guide for information about how to work with Enhanced ShockBurst.
To start developing, check out the :ref:`esb_prx_ptx` sample.

Near field communication
========================

> Near field communication (NFC) is a technology for wireless transfer of small amounts of data between two devices.

Reviewer 7: ```suggestion
Near Field Communication (NFC) is a technology for wireless transfer of small amounts of data between two devices that are in close proximity.
```

It uses very simple communication protocols and therefore allows quicker connection than Bluetooth LE.

> However, because NFC uses magnetic induction to enable communication, the devices must be very close (< 10 cm) to connect.

Reviewer 2: Not sure on this description.  Suggest replace with:
NFC makes transferring data when devices in close proximity fast and easy.  The range of NFC is typically < 10 cm.

> The stack for developing NFC applications is provided in binary format in the `nrfxlib`_ repository.

Reviewer 7: ```suggestion
|NCS| provides 2 protocol stacks for developing NFC applications: Type 2 Tag and Type 4 Tag. These stacks are provided in binary format in the `nrfxlib`_ repository.
```

> See :ref:`nrfxlib:nfc` for documentation about the NFC stack.

Reviewer 7: ```suggestion
See :ref:`nrfxlib:nfc` for documentation about the NFC stacks.
```

> The NFC stack requires the NFCT driver for nRF52 devices, which is available as part of `nrfx`_.

Reviewer 2: The "nrfx" just links out to git hub.  That gives the impression the user needs to do something to use it.  I would prefer we link to a new page, included at the top level "About the nRF Connect SDK" which explains that nrfx contains Nordic's RTOS independent driver implementations, it is already incorporated in the Zephyr repository and is available separately in github <link>."

Author: Will add a sentence that nrfx is included through the Zephyr repo. The other updates should come as part of including the nrfx documentation.

See :ref:`nfc_samples` and :ref:`lib_nfc` for lists of samples and libraries that the |NCS| provides.

USB
===

> The |NCS| uses Zephyr's implementation of a USB device stack.

Reviewer 2: ```suggestion
The |NCS| contains a USB device stack for the USB 2.0 Full Speed peripheral that is available on a number of the nRF52 devices.
```

See :ref:`zephyr:usb_device_stack` for documentation and :ref:`zephyr:usb-samples` for a list of available samples.

> The USB stack requires the USBD driver for nRF52 devices, which is available as part of `nrfx`_.

Reviewer 2: same comment nrfx
